### PR TITLE
DeletedMessagesVault API - support limit query

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
Screenshot from 2023-05-24 14-20-14

**Review by Arsnael — APPROVED** [review]

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/query/QueryDTO.java [hunk 1]
```diff
@@ -36,22 +35,22 @@ public class QueryDTO implements QueryElement {
...
     @VisibleForTesting
     static QueryDTO and(Long limit, QueryElement... queryElements) {
-        return new QueryDTO(Combinator.AND.getValue(), ImmutableList.copyOf(queryElements), Optional.ofNullable(limit));
+        return new QueryDTO(Combinator.AND.getValue(), ImmutableList.copyOf(queryElements), limit);
     }
 
     private final String combinator;
     private final List<QueryElement> criteria;
-    private final Optional<Long> limit;
+    private final Long limit;
```
Comment (chibenwa) [inline mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/query/QueryDTO.java]: Why get rid of the optional? 

Sounds like a "regression" to me...

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/query/QueryDTO.java [hunk 2]
```diff
@@ -36,22 +35,22 @@ public class QueryDTO implements QueryElement {
 
     @VisibleForTesting
     static QueryDTO and(QueryElement... queryElements) {
-        return new QueryDTO(Combinator.AND.getValue(), ImmutableList.copyOf(queryElements), Optional.empty());
+        return new QueryDTO(Combinator.AND.getValue(), ImmutableList.copyOf(queryElements), null);
```
Comment (chibenwa) [inline mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/query/QueryDTO.java:38]: Optionals are better than null for representing absence of something...

**Review by quantranhong1999 — COMMENTED** [review]
Can you make the static variable `DeletedMessagesVaultRestoreTask.TYPE` public also? Thanks in advanced :-)

**Review by chibenwa — APPROVED** [review]

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/query/QueryDTO.java [hunk 3]
```diff
@@ -65,6 +67,13 @@ public List<QueryElement> getCriteria() {
         return criteria;
     }
 
+
+    @JsonGetter("limit")
+    public Long getLimitAsLong() {
+        return limit.orElse(null);
+    }
+
+    @JsonIgnore
```
Comment (chibenwa) [inline mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/query/QueryDTO.java:76]: Add the JDK 8 Jackson Module to the serializer instead?

Comment (vttranlina) [reply mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/query/QueryDTO.java:76]: Can you explain more? 

`@JsonIgnore` just for ignoring the `Optional limit` 

> Add the JDK 8 Jackson Module 

Your mean is `new ObjectMapper().registerModule(new Jdk8Module());` ? , the `QueryElementSerializer` already receives the `ObjectMapper` bean in constructors, I don't think we should modify the bean

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**